vertx-web/src/main/java/io/vertx/ext/web/templ/TemplateEngine.java

```diff
@@ -33,12 +33,17 @@
 public interface TemplateEngine {
 
   /**
-   * Render
+   * Render the template
+   * <p>
+   * <b>NOTE</b> if you call method directly (i.e. not using {@link io.vertx.ext.web.handler.TemplateHandler}) make sure
+   * that <i>templateFileName</i> is sanitized via {@link io.vertx.ext.web.impl.Utils#normalizePath(String)}
+   *
    * @param context  the routing context
-   * @param templateFileName  the template file name to use
+   * @param templateDirectory  the template directory to use
+   * @param templateFileName  the relative template file name to use
    * @param handler  the handler that will be called with a result containing the buffer or a failure.
    */
-  void render(RoutingContext context, String templateFileName, Handler<AsyncResult<Buffer>> handler);
+  void render(RoutingContext context, String templateDirectory, String templateFileName, Handler<AsyncResult<Buffer>> handler);
```

Comment:
When extracting `templateFileName` had to do `templateFileName.substring(sep),` without  _+ 1_ since template engine expects the filename to have leading / and removes it (we discussed this earlier when we put in javadoc to be run thru `Utils#normalizePath(String)`)